### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Transition Experience, tbd, V1 Transition Experience

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#88272]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** There are some places in your designs that don't have headings identified but I think are important to code as headings. This might already be on your radar, but I'll note them just to be safe:
   - On [Getting started with VA](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=TK0Zba25FMUxWThX-4), "Try our new recommendation tool" should be an H2
   - On [VA benefits for service members](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), "Try our new recommendation tool" should be an H2 (and also a link, matching the same pattern on [VA health care](https://www.va.gov/health-care/)).
   - On [Review your entries](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), each accordion open/close button should also be an H2.
- [x] **Must:** On the [intro page for the recommendation tool](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), it looks like the "Get started" button is navigational. This should be a link rather than a button.
- [x] **Should:** On the [Goals](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4) step, the note between the checkboxes and the back/continue buttons is unlikely to be noticed by screen reader users. When going through a form flow, screen reader users typically don't have their computers announce everything on the page. Instead, they'll press [tab] to skip between interactive elements --- meaning anything not programmatically associated with an interactive element won't be announced to them.
   
   There are a few options for programmatically associating that text with an interactive element, but I think the easiest option would be to add it to the hint text on the `va-checkbox-group` component, eg. "Check all that apply. This is a new tool on VA.gov. The current version..." Alternatively, consider removing that line altogether. It doesn't seem particularly critical to the user flow.
- [x] **Must:** On the [Service](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4) step, split this into two pages following the [single response/one thing per page](https://design.va.gov/patterns/ask-users-for/a-single-response) guidance in the design system. They can both still be part of Step 2, but something like:
   Step 2 [first page]
      - In total, how long have you served in the military?
      - Are you currently serving in the military?
   Step 2 [second page, conditional if "Are you currently...?" is "Yes"]
       - [H3 heading to differentiate it from the previous page --- something like "Your future separation" or whatever content makes the most sense]
       - When do you expect to separate or retire from service?
       - [conditionally revealed if answer is "I know the month/year"] What is your expected separation date?
   I think that should be consistent with the [conditional reveal guidance](https://design.va.gov/patterns/ask-users-for/a-single-response#conditionally-revealed-fields), and I think groups topics in a way that's clear. 
- [x] **Coding note:** For the [Discharge](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4) and [Disability](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4) questions, I believe the `hint` prop for the form components only supports plain text --- so no links. Double-check that to make sure, but you may need to rearrange things a little. Note that from an accessibility standpoint I think it would be fine to put both of those links after the questions/before the back/continue buttons. Since links are interactive elements, they'll receive focus and won't be missed by screen reader users tabbing through the page.
- [x] **Coding note:** On [Review your entries](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), note that each Edit button will need an `aria-label` to help screen reader users distinguish between them, or the `label` prop on the `va-button` component`. Something like `Edit Goals` and `Edit Your military service` is perfect. 
- [x] **Must:** In the [save, bookmark, and share modal](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), make sure you have a clear plan for what happens after you click each of those buttons --- and how it's announced to screen reader users. My recommendation would be using a [slim alert](https://design.va.gov/components/alert#examples---slim-alert) displaying immediately below each option ("URL has been copied to your clipboard" success alert, for example), with keyboard focus set to a `div` containing the alert. The alert ensures that users know something has happened and it worked or didn't work, and moving keyboard focus ensures that the alert is immediately announced by screen readers.
- [x] **Must:** In the [save, bookmark, and share modal](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), note that the heading should be an H2, not an H1. This should be taken care of for you if you're using the `va-modal` component.
- [x] **Should:** On [Your benefits and resources](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), time-sensitive benefits are hidden within a collapsed expandable alert. I would recommend moving them out of the expandable alert, and just having them in either a regular info alert or as plain content on the page. Expandable alerts don't have a heading associated with them, which may make it more difficult for screen reader users to discover that section compared to others. I also feel like if it's time-sensitive you want to be drawing extra attention to it, not keeping it tidy and compact.
- [x] **Should:** On the [Your benefits and resources](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4) page, I'd recommend grouping "Filter by type," "Sort results by," and any other filter/sort options you include into a single "filter and sort" section --- with a single button ("Update results" or "Filter and sort" or similar) that applies all of them at once. Clicking that button should move keyboard focus to the "Showing X results" message, and combining it all into a single button means screen reader users won't accidentally skip over an option that they care about.
- [x] **Should:** I would recommend thinking about reorganizing the results on [Your benefits and resources](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4) so that the filter/sort options appear beneath the "Recommended benefits and resources" heading, or otherwise beneath a heading that describes the scope of what's being filtered/sorted. If I'm a screen reader user, the filter/sort section exists a little outside of any clear context, and it's not totally clear that I'd discover those options if I skipped straight to the "Recommended benefits and resources" section.
- [x] **Must:** On [Your benefits and resources](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), the accordion open/close buttons and card headings all need to be coded as headings, nested such that the structure of the page is clear. What each heading level is will really depend on if/how you restructure this page.
- [x] **Coding note:** For each card, I would recommend coding the tag + heading text together as:
   ```
   <hx>
      <span class="usa-label any-other-classes-needed-to-style-appropriately">Education</span>
      <span class="classes-for-regular-heading-text">GI Bill</span>
   </hx>
   ```
   This will ensure the context provided by the tag doesn't get lost for screen reader users skipping to that heading.
- [x] **Coding note:** On [Your benefits and resources](https://www.figma.com/design/DNy6D8ELfTQW1g3EK7yJu5/Mid-Fidelity-Prototype-for-v1?node-id=2683-30694&t=M5XcnlUE8IzraEPk-4), each "Learn more" and "Apply now" link will need an `aria-label` (or `label` prop on the `va-link` component) to ensure each one can be distinguished by screen reader users with a unique accessible name. Something like `Learn more about GI Bill benefits` and `Apply now for GI Bill benefits` or similar would be great.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone
